c++/src/ColumnReader.cc

```diff
@@ -1553,6 +1558,8 @@ namespace orc {
     ColumnReader::seekToRowGroup(positions);
     valueStream->seek(positions.at(columnId));
     scaleDecoder->seek(positions.at(columnId));
+    buffer = nullptr;
+    bufferEnd = nullptr;
```

Comment:
Just a question: The changes in this file look general instead of specific to this PPD. Previously, we don't need these?

Reply:
Then, can we have this change first as a separate PR?